### PR TITLE
Fix loading incomplete NFTs 

### PR DESCRIPTION
**Pull request by Jibz-Mysten**
## Description 

Describe the changes or additions included in this PR.

- Fix wallet loading partial NFT by fetching and filtering for NFTs (10). Scroll to load more 
- Move `useGetOwnedObjects` and `useOnScreen` to core
- Added map key to BalanceChangeEntry

## Test Plan 

https://user-images.githubusercontent.com/126525197/235388841-bae92114-d54b-4efd-87f0-7afbedb64478.mov

How did you test the new or updated feature?

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [x] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: apps/wallet/src/ui/app/pages/home/nfts/index.tsx [hunk 1]
```diff
@@ -1,26 +1,75 @@
...
+    const { isIntersecting } = useOnScreen(observerElem);
+    const isSpinnerVisible = isFetchingNextPage && hasNextPage;
+
+    const nfts =
+        data?.pages
+            .flatMap((page) => page.data)
+            .filter((resp) => !!getObjectDisplay(resp).data)
+            .map(({ data }) => data as SuiObjectData) || [];
+
+    useEffect(() => {
```
Comment (plam-ml) [inline apps/wallet/src/ui/app/pages/home/nfts/index.tsx:49]: should we group these 2 `useEffect` since they both are calling fetchNextPage?

**Review by plam-ml — APPROVED** [review]
lgtm with 1 q

File: apps/core/src/hooks/useGetOwnedObjects.ts [hunk 2]
```diff
@@ -1,29 +1,30 @@
...
+            rpc.getOwnedObjects({
                 owner: address!,
                 filter,
                 options: {
                     showType: true,
                     showContent: true,
                     showDisplay: true,
                 },
-                limit: MAX_OBJECTS_PER_REQ,
+                limit: maxObjectRequests,
```
Comment (Jordan-Mysten) [inline apps/core/src/hooks/useGetOwnedObjects.ts:27]: Needs to be part of the cache key

File: apps/wallet/src/ui/app/pages/home/nfts/index.tsx [hunk 3]
```diff
@@ -1,26 +1,76 @@
...
+        if (
+            (isIntersecting && hasNextPage && !isFetchingNextPage) ||
+            (nfts.length < DEFAULT_NFT_LIMIT &&
```
Comment (Jordan-Mysten) [inline apps/wallet/src/ui/app/pages/home/nfts/index.tsx:52]: I probably wouldn't do this special case. The onscreen observer should handle the case where there's not enough to populate one page. I think that's enough.

File: apps/wallet/src/ui/app/pages/home/nfts/index.tsx [hunk 4]
```diff
@@ -1,26 +1,76 @@
...
-    const nfts = data?.data
-        ?.filter((resp) => !!getObjectDisplay(resp).data)
-        .map(({ data }) => data as SuiObjectData);
+    const observerElem = useRef<HTMLDivElement | null>(null);
+    const { isIntersecting } = useOnScreen(observerElem);
+    const isSpinnerVisible = isFetchingNextPage && hasNextPage;
+
+    const nfts =
+        data?.pages
+            .flatMap((page) => page.data)
```
Comment (Jordan-Mysten) [inline apps/wallet/src/ui/app/pages/home/nfts/index.tsx:45]: It's so rare that I see a good usecase for flatmap, this is a good one!

**Pull request merged**